arbnode/inbox_reader.go

```diff
@@ -139,6 +139,9 @@ func (r *InboxReader) Start(ctxIn context.Context) error {
 			return err
 		}
 		if batchCount > 0 {
+			if r.tracker.snapSyncConfig.Enabled {
```

Comment:
flagging for my self:
I think it's fine because we can validate chainConfig against arbos-stored chain config - but  need to talk to Lee about that.